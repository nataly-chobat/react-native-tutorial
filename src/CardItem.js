import * as React from 'react';
import { View, Text, StyleSheet, Image, TouchableOpacity, Animated, Easing, Button } from 'react-native';

class CardItem extends React.PureComponent {
    state = {
        scaleValue: new Animated.Value(0),
        translateX: new Animated.Value(-500),
        translateY: new Animated.Value(100),
    }

    componentDidMount(){
        const { scaleValue, translateX, translateY } = this.state;
        const { id } = this.props;

        Animated.timing(scaleValue, {
            toValue: 1,
            duration: 1000,
            delay: id * 300,
            nativeDriver: true,
        }).start();

        Animated.timing(translateX, {
            toValue: 0,
            duration: 500,
            easing: Easing.bezier(0.19, 0.32, 0, 0.97),
            delay: id * 300,
        }).start();

        Animated.timing(translateY, {
            toValue: 0,
            duration: 500,
            easing: Easing.bezier(0.19, 0.32, 0, 0.97),
            delay: id * 300,
        }).start();
    }

    navigateToItemDetail = () => {
        const { id, name, navigateToItemDetail } = this.props;

        navigateToItemDetail(id, name);
    }

    render() {
        const { scaleValue, translateX, translateY } = this.state;
        const { name, description, breed, url } = this.props;

        return (

                <Button testID="catInfo" onPress={this.navigateToItemDetail}>
                    <View style={styles.container}>
                        <View style={styles.mainInfo}>
                            <Text style={styles.heading}>{name}</Text>
                            <Text style={styles.breed}>{breed}</Text>
                        </View>
                        <View style={styles.line}/>
                        <View style={styles.imageWrapper}>
                            <Image style={styles.image} source={{uri: url}} />
                        </View>
                        <Text numberOfLines={2} ellipsizeMode="tail">{description}</Text>
                    </View>
                </Button>
        )
    }
}

export default CardItem;

const styles = StyleSheet.create({
    container: {
        borderWidth: 1,
        borderColor: '#ccc',
        borderRadius: 5,
        flexDirection: 'column',
        margin: 10,
        paddingVertical: 15,
        paddingHorizontal: 10
    },
    heading: {
        fontWeight: 'bold',
        fontSize: 14
    },
    line: {
        borderBottomWidth: 1,
        marginVertical: 10,
        borderColor: '#d1d1d1'
    },
    mainInfo: {
        alignItems: 'center',
        justifyContent: 'space-between',
        flexDirection: 'row'
    },
    image: {
        width: 260,
        height: 130,
        marginVertical: 10
    },
    imageWrapper: {
        alignItems: 'center',
    }
});