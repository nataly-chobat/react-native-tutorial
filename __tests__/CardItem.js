import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import CardItem from '../src/CardItem';

const defaultProps = {
    id: 1,
    name: 'Габи',
    url: 'https://petstory.ru/resize/800x800x80/upload/images/articles/breeds/abissinskaya-koshka/abisin_cat_3.jpg',
    breed: 'Абиссинская кошка',
    description: 'Эти кошки становятся похожими на пантеру, когда начинают играть или чем-то занимаются.',
    age: '5 месяцев',
    randomFact: 'Cамый тяжелый когда-либо зарегистрированный кот весил около 20 кг!',
    navigateToItemDetail: jest.fn(),
};

describe('CardItem component', () => {
    let component;

    beforeEach(() => {
        component = shallow(<CardItem {...defaultProps} />);
    });

    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('calls prop navigateToItemDetail when Button clicked', () => {
        component
            .find({
                testID: 'catInfo',
            })
            .simulate('press');

        expect(defaultProps.navigateToItemDetail).toHaveBeenCalled();
    });
});


// Note: test renderer must be required after react-native.
// import renderer from 'react-test-renderer';
//
// it('renders correctly', () => {
//     renderer.create(<CardItem />);
// });
