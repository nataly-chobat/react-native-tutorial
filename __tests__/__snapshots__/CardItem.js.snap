// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItem component renders correctly 1`] = `
<Button
  onPress={[Function]}
  testID="catInfo"
>
  <View
    style={
      Object {
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "column",
        "margin": 10,
        "paddingHorizontal": 10,
        "paddingVertical": 15,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        Габи
      </Text>
      <Text>
        Абиссинская кошка
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#d1d1d1",
          "marginVertical": 10,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Component
        source={
          Object {
            "uri": "https://petstory.ru/resize/800x800x80/upload/images/articles/breeds/abissinskaya-koshka/abisin_cat_3.jpg",
          }
        }
        style={
          Object {
            "height": 130,
            "marginVertical": 10,
            "width": 260,
          }
        }
      />
    </View>
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
    >
      Эти кошки становятся похожими на пантеру, когда начинают играть или чем-то занимаются.
    </Text>
  </View>
</Button>
`;
